frodo.js

```diff
@@ -3,24 +3,29 @@
  *
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.